src/main/js/pullRequest.ts

```diff
@@ -39,27 +143,277 @@ export function createPullRequest(url: string, pullRequest: BasicPullRequest) {
       return apiClient.get(location + "?fields=id");
     })
     .then(response => response.json())
-    .then(pr => pr.id);
+    .then(pr => {
+      if (callback) {
+        callback(pr.id);
+      }
+      return pr;
+    });
 }
 
-export function updatePullRequest(url: string, pullRequest: PullRequest) {
-  return apiClient.put(url, pullRequest, CONTENT_TYPE_PULLREQUEST);
+export const useRejectPullRequest = (repository: Repository, pullRequest: PullRequest) => {
+  const queryClient = useQueryClient();
+  const { mutate, isLoading, error } = useMutation<{}, Error, PullRequest>(pr => rejectPullRequest(pr), {
+    onSuccess: () => {
+      return queryClient.invalidateQueries(prQueryKey(repository, pullRequest.id!));
```

Comment:
A much better way is to check and throw an error at the top of the hook instead of adding an `!` and hope for the best.

Reply:
This invalidates only the detail queries, not the one for the list.